libbpf-rs/src/perf_buffer.rs

```diff
@@ -104,29 +120,32 @@ impl<'a, 'b> PerfBufferBuilder<'a, 'b> {
             ));
         }
 
-        let c_sample_cb: libbpf_sys::perf_buffer_sample_fn = if self.sample_cb.is_some() {
-            Some(Self::call_sample_cb)
-        } else {
-            None
-        };
-
-        let c_lost_cb: libbpf_sys::perf_buffer_lost_fn = if self.lost_cb.is_some() {
-            Some(Self::call_lost_cb)
-        } else {
-            None
-        };
-
         let callback_struct_ptr = Box::into_raw(Box::new(CbStruct {
             sample_cb: self.sample_cb,
             lost_cb: self.lost_cb,
         }));
 
+        let mut attr = unsafe {
+            libbpf_sys::perf_event_attr {
+                type_: libbpf_sys::PERF_TYPE_SOFTWARE,
+                config: libbpf_sys::PERF_COUNT_SW_BPF_OUTPUT as u64,
+                sample_type: libbpf_sys::PERF_SAMPLE_RAW as u64,
+                __bindgen_anon_1: libbpf_sys::perf_event_attr__bindgen_ty_1 {
+                    sample_period: self.wakeup_events as u64,
```

Comment:
Not really familiar with the perf event API and documentation seems non-existent: why are we setting a sampling period here? Can you please clarify and add a comment?

Reply:
@arilou if this is useful option, it probably would be better to add it on libbpf side into perf_buffer_opts, and then keep using non-raw variant in libbpf-rs. Would you be able to submit a patch to bpf@vger.kernel.org for this?

Reply:
I think I'll remove the wakeup_events patch from this PR and after ill send a patch to libbpf and libbpf-sys will move to where that patch is, I'll re-submit a new PR

Reply:
Sounds good, thanks!